### PR TITLE
graphql server creation #1015

### PR DESCRIPTION
**Pull request by UmmulkiramR**
added 
- a gql server
- a gql schema
- a resolver for fetching clinical registration data

### CODE REVIEW

**Comment by demariadaniel** [comment]
Can you please connect Issue to PR?

File: package.json [hunk 1]
```diff
@@ -110,7 +114,7 @@
     "deep-freeze": "^0.0.1",
     "dotenv": "^8.0.0",
     "errorhandler": "^1.5.1",
-    "express": "^4.17.1",
+    "express": "4.17.0",
```
Comment (demariadaniel) [inline package.json:117]: Is there a reason we are not using compatible patch versions here?

File: src/dictionary/manager.ts [hunk 2]
```diff
@@ -200,7 +200,8 @@ class SchemaManager {
       return newSchema;
     } catch (err) {
       L.error('Failed to fetch schema: ', err);
-      throw new Error('Failed to fetch schema: ' + err.message);
+      throw new Error('Failed to fetch schema: ' + (err as Error).message); // added 'as Error' due to Typescript version upgrade
+      throw new Error('Failed to fetch schema: ');
```
Comment (demariadaniel) [inline src/dictionary/manager.ts:204]: This line seems like a duplicate / unnecessary?

File: src/submission/submission-api.ts [hunk 3]
```diff
@@ -50,6 +50,16 @@ class SubmissionController {
     return res.status(200).send(registration);
   }
 
+  async getRegistrationDataByProgramId(programId: string) {
+    L.debug('in getRegistrationByProgramId');
+    // const programId = programId;  // req.params.programId;
+    const registration = await submission.operations.findByProgramId(programId);
+    /*if (registration == undefined) {
+      return res.status(200).send({});
+    }*/
```
Comment (demariadaniel) [inline src/submission/submission-api.ts:59]: Comments here as well

**Comment by demariadaniel** [comment]
A few minor clean up items but this is working for me, great job!

File: package.json (hunk 1 above)
Comment (UmmulkiramR) [reply package.json:117]: yes, the Apollo server library pulls its own express dependency. Want to avoid conflicts due to that.

**Review by demariadaniel — APPROVED** [review]

**Pull request merged**